keras/src/callbacks/hard_terminate_on_nan_test.py

```diff
@@ -0,0 +1,186 @@
+"""Tests for TerminateOnNaN callback."""
+
+import os
+import tempfile
+
+import numpy as np
+import pytest
+
+import keras
+from keras.src import backend
+from keras.src import layers
+from keras.src import models
+from keras.src import testing
+from keras.src.callbacks import BackupAndRestore
+from keras.src.callbacks import TerminateOnNaN
+
+
+@pytest.mark.skipif(
+    backend.backend() in ["numpy", "openvino"],
+    reason="TerminateOnNaN not supported for NumPy or OpenVINO backend",
+)
+class TerminateOnNaNTest(testing.TestCase):
```

Comment:
There are some tests in `keras/src/callbacks/terminate_on_nan_test.py` already.

Please move the new tests there and combine them.